### PR TITLE
ice cream machines now contain and consume resources

### PR DESCRIPTION
**Pull request by MadmanMartian**
Rather than creating phantom reagents from nowhere, now the ice cream machine draws from an internal container.

You can add back to this container with a beaker.

TODO

- [x] Make sure you can't just reload the reagents by deconstructing and reconstructing the machine

**Related PRs to the kitchen rework:**

- #16820 Reworks the Ice cream machine to contain and use resources
- #16812 Removes the weird and wonderful goofkitchen machines
- #16829 Adds an actual oven

:cl:
* rscadd: Ice cream machine now draws from an internal reservoir rather than drawing nutrient from the aether
* rscadd: To reload the ice cream machine, put a beaker of the same reagent type in and press add amount

### CODE REVIEW

**Comment by chesse20** [comment]
this ruins the point of the ice cream maker though

**Comment by DamianX** [comment]
What's the point of the ice cream maker?

**Comment by chesse20** [comment]
free food generator for when theres no chef

**Comment by DamianX** [comment]
my immersion

**Comment by NigglyWiggly** [comment]
Since when was that the point of the ice cream maker?

Isn’t the point of it to make ice cream?

File: code/modules/food/icecreamvat.dm [hunk 1]
```diff
@@ -12,51 +12,73 @@
...
+	to_add = null
+	..()
 
 // Utilities ///////////////////////////////////////////////////
 
+/obj/machinery/cooking/icemachine/proc/generate_icecream_reagents(var/reagent_total)
+	reagents.add_reagent(soda, min(reagent_total-reagents.get_reagent_amount(soda), reagent_total))
+	reagents.add_reagent(alcohol, min(reagent_total-reagents.get_reagent_amount(alcohol), reagent_total))
+	reagents.add_reagent(CREAM, min(reagent_total-reagents.get_reagent_amount(CREAM), reagent_total))
+	reagents.add_reagent(WATER, min(reagent_total-reagents.get_reagent_amount(WATER), reagent_total))
```
Comment (Intigracy) [inline code/modules/food/icecreamvat.dm]: this should probably be ICE

Comment (Pathid) [reply code/modules/food/icecreamvat.dm]: Maybe so, but ice dispensing is broken right now. #17029

**Comment by DamianX** [comment]
Inactive for a month, he's busy with antag datums
Feel free to reopen

**Comment by Intigracy** [comment]
ICE

File: code/modules/food/icecreamvat.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/modules/food/icecreamvat.dm]: It was water, but it's fixed up.

**Comment by ShiftyRail** [comment]
Why was this merged

**Comment by io-june** [comment]
Well it was finished recently so why not?
This is pretty old PR, probably it was discussed few times earlier in the discord

**Comment by bathosbathtime** [comment]
I've been informed by both the coder d*scord and the thread that there is an additional reason for this. There have been cases of users producing enough ice cream with the machine that it was causing clientside crashes.  I have not looked into this at all to confirm that it is true but I feel for the sake of transparency that if that was a reason this was merged it should be included on the pull request itself.

**Pull request merged**